src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -428,7 +433,8 @@ nf_task_exit() {
 }
 
 nf_task_entry() {
-  if [ -n "$docker_creds" ]; then
+  docker_credentials=$(dx find data --path "$DOCKER_CREDS_FOLDER" --name "$DOCKER_CREDS_FILENAME")
```

Comment:
How can we avoid this API call, esp since it doesn't qualify the search with a project/container? Is it not enough to check if the docker_creds input was provided (`if [ -n "$docker_creds" ]; then`)? Was the motivation to check that this input file was indeed downloaded into the workspace? But if it was not, that means that something went wrong anyway (the user provided the creds but they are not in the workspace) so perhaps it's better to throw an error here early?

Reply:
No really, this is task entry and it does not have docker_creds in the input, this is the reason why I saved the file to the container.

With Yih-Chii, the general premise is not to polute the input space of subjob.

Reply:
Btw this search is done only within a container, isnt it? Otherwise --all-projects would have to be selected, right? We also search only through certain folder - which should reduce the weight of this api call

Reply:
It could be either done with the container context or the project context, so this is either something to be checked or, if possible, if you could qualify the search space with WORKSPACE_ID?

Reply:
Added DX_WORKSPACE_ID to the search.